sunpy/net/dataretriever/client.py

```diff
@@ -225,7 +170,57 @@ def _can_handle_query(cls, *query):
         `sunpy.net.fido_factory.UnifiedDownloaderFactory`
         class uses to dispatch queries to this Client.
         """
-        raise NotImplementedError
+        adict = cls.register_values()
+        if cls.optional == set():
+            cls.optional = {k for k in adict.keys()} - cls.required
+        if not cls.check_attr_types_in_query(query, cls.required, cls.optional):
+            return False
+        for key in adict:
+            all_vals = [i[0].lower() for i in adict[key]]
+            for x in query:
+                if isinstance(x, key) and str(x.value).lower() not in all_vals:
+                    return False
+        return True
+
+    def post_search_hook(self, exdict, matchdict):
+        """
+        Helper function used after `:func:~sunpy.net.dataretriever.GenericClient.search`
+        which makes the extracted metadata representable in a query response table.
+
+        Parameters
+        ----------
+        exdict: `dict`
+            Represents metadata extracted from files.
+        matchdict: `dict`
+            Contains attr values accessed from `register_values()`
+            and the search query itself.
+
+        Returns
+        -------
+        map_: `~collections.OrderedDict`
+            An Ordered Dictionary which is used by `QueryResponse`
+            to show results.
+        """
+        map_ = OrderedDict()
+        almost_day = TimeDelta(1 * u.day - 1 * u.millisecond)
+        if 'month' in exdict and 'day' in exdict:
+            start = parse_time("{}/{}/{}".format(exdict['year'], exdict['month'], exdict['day']))
+            end = start + almost_day
+        elif 'year' in exdict:
+            start = parse_time("{}/{}/{}".format(exdict['year'], 1, 1))
+            end = parse_time("{}/{}/{}".format(exdict['year'], 12, 31)) + almost_day
+        map_['Time'] = TimeRange(start, end)
+        map_['Start Time'] = start.strftime(TIME_FORMAT)
+        map_['End Time'] = end.strftime(TIME_FORMAT)
+        map_['Instrument'] = matchdict['Instrument'][0].upper()
+        if 'Physobs' in matchdict:
+            map_['Physobs'] = matchdict['Physobs'][0]
+        map_['Source'] = matchdict['Source'][0].upper()
+        map_['Provider'] = matchdict['Provider'][0].upper()
+        for k in exdict:
+            if k not in ['year', 'month', 'day']:
+                map_[k] = exdict[k]
+        return map_
 
     def _get_full_filenames(self, qres, filenames, path):
         """
```

Comment:
This docstring is telling lies, it downloads nothing.

Reply:
Yes, updated.